specutils/tests/test_analysis.py

```diff
@@ -1020,27 +1021,17 @@ def test_moment_cube():
     # spatial 1st order - returns the dispersion
     moment_1 = moment(spectrum, order=1, axis=1)
 
-    assert moment_1.shape == (10,10000)
+    assert moment_1.shape == (9, 10000)
     assert moment_1.unit.is_equivalent(u.GHz)
     assert quantity_allclose(moment_1, frequencies, rtol=1.E-5)
 
     # higher order
     moment_2 = moment(spectrum, order=2)
 
-    assert moment_2.shape == (10,10)
+    assert moment_2.shape == (9, 10)
     assert moment_2.unit.is_equivalent(u.GHz**2)
     assert quantity_allclose(moment_2, 816.648*u.GHz**2, atol=0.01*u.GHz**2)
 
-    # spatial higher order (what's the meaning of this?)
-    moment_2 = moment(spectrum, order=2, axis=1)
```

Comment:
This no longer works with modified test case, some failure broadcasting shapes. And since the author does not know what is the meaning of this usage, might as well remove it?

Reply:
This test case definitely needs to work!  I think the author is saying it's not clear this has *physical* meaning, but I do have one: a collection of spectra from one instrument (e.g. a fiber spectrograph or some IFUs that are "slicers") have this same shape (nfiber, nspatial, nspectral), and sometimes one wants to know, say, where a trace is in the spatial axes.  So that's the order=1, axis=1 case.

More to the point, if this isn't working that means something's not quite broadcasting right in the implementation... So I think this case needs to be addressed (I can't vouch for the *values*, although I assume whoever wrote this checked them.  But I think the shape check should still be there, just that it should be (9, 10000) in the changed shape

Reply:
I don't think this test case of `order=2, axis=1` ever worked when your spatial-spatial slice is not a square, regardless of my fix or not. In fact, the logic of this test case does not even go through my fix because of the `axis` check.

Reply:
Ooh, gotcha.  yes, then this can merge then now that you point out it's definitely unrelated to this fix.  But in definitely *should* work for non-square axes!  So I'll make an issue to track that down.